isis/tests/XmlToJsonTests.cpp

```diff
@@ -0,0 +1,117 @@
+      </First>
+      <First>
+        <tweleve>12</tweleve>
+        <thirteen>13</thirteen>
+      </First>
+      <First>
+        <fourteen>14</fourteen>
+        <fifteen>15</fifteen>
+      </First>
+      <Second>
+          <A attributeA="A" attributeB="B" attributeC="C">ElementValue</A>
+      </Second>
+      <Third>
+        <Greek attr="alphabet" otherattr="firstLetter" >
+          <Alpha attr1="attr1" attr2="attr2">AlphaValue</Alpha>
+          <Beta attrbeta1="beta1" attrbeta2="beta2">BetaValue</Beta>
+          <AnotherLevel>
+            <Gamma attrgamma="gamma">GammaValue</Gamma>
+          </AnotherLevel>
+        </Greek>
+      </Third>
+  </TagLevel1B>
+</TagLevel0>)";
+
+  QDomDocument xmlDocument("TestDocument");
+  xmlDocument.setContent(xmlInput);
+  ordered_json result = xmlToJson(xmlDocument);
+  std::cout << result.dump(4) << std::endl;
+
+  // Test deeply nested value retrieval (uncomplicated)
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2B"], "TagLevel2BValue");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2C"]["TagLevel3"]["TagLevel4A"], "TagLevel4AValue");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2C"]["TagLevel3"]["TagLevel4C"]["TagLevel4D"]["TagLevel4E"], "DeepValue");
+
+  // Test attributes (uncomplicated)
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2D"]["@attributeTag2D"], "Attribute value");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2D"]["#text"], "TagLevel2DValue");
+
+  // Test no-text value cases <tag /> and <tag attributeName="attributeValue" /> 
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2ExtraExtra"], nullptr);
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1A"]["TagLevel2Extra"]["@attr"], "justAnAttribute");
+
+  // Test list creation for repeated tags at the same level
+  
+  // Case A: <a><b>bcontents</b></a> <a><c>cContents</c></a> JSON: a: [ {b: bcontents}, {c: cContents} ]
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][1]["tweleve"], "12");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][1]["thirteen"], "13");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][2]["fourteen"], "14");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][2]["fifteen"], "15");
+
+  // Case B: <z><a>aContents1</a><a>aContents2</a></a> JSON: z: {a: [aContents1, aContents2] }
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["A"][0], "A1");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["A"][1], "A2");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["oddball"], "notrepeated");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["A"][2]["B"][0], "b1");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["A"][2]["B"][1], "b2");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["A"][2]["C"], "notlist");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["ten"][0], "10");
+  EXPECT_EQ(result["TagLevel0"]["TagLevel1B"]["First"][0]["ten"][1], "TEN");
```

Comment:
This test is way too big. If you are commenting that there are multiple test cases, you should split the test up.

Reply:
Yeah. I think I'm going to keep some kind of "megatest" to make sure everything works together, but also add tests for each subsection I'm currently testing in the single monolithic test.